lib/src/io_utils.dart

```diff
@@ -118,40 +120,102 @@ final RegExp partOfRegexp = RegExp('part of ');
     'as Dartdoc 1.0.0')
 final RegExp newLinePartOfRegexp = RegExp('\npart of ');
 
-/// Best used with Future<void>.
-class MultiFutureTracker<T> {
-  /// Approximate maximum number of simultaneous active Futures.
-  final int parallel;
+typedef TaskQueueClosure<T> = Future<T> Function();
+
+class _TaskQueueItem<T> {
+  _TaskQueueItem(this._closure, this._completer, {this.onComplete});
+
+  final TaskQueueClosure<T> _closure;
+  final Completer<T> _completer;
+  void Function() onComplete;
+
+  Future<void> run() async {
+    try {
+      final result = await _closure();
+      if (result != null) {
+        _completer.complete(result);
+      } else {
+        _completer.complete(null);
+      }
+      await Future<void>.microtask(() {});
```

Comment:
Would you mind adding a comment about why this is needed?

Reply:
Actually, I'm not sure it is.  I put it there to prevent other futures from running until it completes, but I don't think it really does what I think it does.  I'm running the tests again to see if removing it makes any difference.

Reply:
Yeah, looks like I don't need it.